### PR TITLE
mame: init at 0.215

### PR DESCRIPTION
**Pull request by ghost**
###### Motivation for this change

MAME is a multi-purpose emulation framework.

Closes #36206

MAME’s purpose is to preserve decades of software history. As electronic technology continues to rush forward, MAME prevents this important "vintage" software from being lost and forgotten. This is achieved by documenting the hardware and how it functions. The source code to MAME serves as this documentation. The fact that the software is usable serves primarily to validate the accuracy of the documentation (how else can you prove that you have recreated the hardware faithfully?). Over time, MAME (originally stood for Multiple Arcade Machine Emulator) absorbed the sister-project MESS (Multi Emulator Super System), so MAME now documents a wide variety of (mostly vintage) computers, video game consoles and calculators, in addition to the arcade video games that were its initial focus.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @

### CODE REVIEW

**Comment by gnidorah** [comment]
@GrahamcOfBorg build mame

File: pkgs/misc/emulators/mame/default.nix [hunk 1]
```diff
@@ -0,0 +1,59 @@
+{ stdenv, mkDerivation, fetchFromGitHub, makeDesktopItem
+, python, pkgconfig, SDL2, SDL2_ttf, alsaLib, which, qtbase, libXinerama }:
+
+let
+  majorVersion = "0";
+  minorVersion = "215";
+
+  desktopItem = makeDesktopItem {
+    name = "MAME";
+    exec = "mame${if stdenv.is64bit then "64" else ""}";
```
Comment (worldofpeace) [inline pkgs/misc/emulators/mame/default.nix]: How about `optionalString stdenv.is64Bit "64"`?

File: pkgs/misc/emulators/mame/default.nix [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+    rev = "mame${majorVersion}${minorVersion}";
+    sha256 = "1phz846p3zzgzrbfiq2vn79iqar2dbf7iv6wfkrp32sdkkvp7l3h";
+  };
+
+  hardeningDisable = [ "fortify" ];
+  NIX_CFLAGS_COMPILE = [ "-Wno-error=maybe-uninitialized" ];
+
+  makeFlags = [ "TOOLS=1" ];
+
+  buildInputs = [ SDL2 SDL2_ttf alsaLib qtbase.dev libXinerama ];
```
Comment (worldofpeace) [inline pkgs/misc/emulators/mame/default.nix]:
```suggestion
  buildInputs = [ SDL2 SDL2_ttf alsaLib qtbase libXinerama ];
```

shouldn't need to specify outputs.

**Review by worldofpeace — CHANGES_REQUESTED** [review]

**Comment by gnidorah** [comment]
@worldofpeace Done, thank you!

**Review by worldofpeace — APPROVED** [review]
LGTM. Thanks :sparkles:

**Comment by gnidorah** [comment]
@worldofpeace Thank you!

**Pull request merged**